### DISCUSSION TITLE
Résultats du poids du site - inc

### DISCUSSION THREAD

**Discussion by DocRoms** [cnumr/EcoIndex#144] [Bug]
Lorsque je teste cette page : https://fr.euronews.com/green/2022/09/16/en-angola-le-deminage-garantit-securite-et-developpement-economique (qui est "moyenne") j'obtient des résultats assez différents sur Ecoindex. 

En analysant de mon coté, en navigation privée, j'obtient un poids, après scroll de 1.9Mo, 1.5 à 1.7Mo sans scroll.
![BugPoidsEcoIndexBrowser](https://user-images.githubusercontent.com/12815769/190845690-37efe524-d293-472a-875e-da9ee0cb3988.png)

En testant sur le site webpagetest.org (étant la référence des sites de webperf) j'obtient une taille de 1.5 mo, qui correspond à peut près à mes tests en navigation privée.
![BugPoidsEcoIndexWebPageTest](https://user-images.githubusercontent.com/12815769/190845717-77cd0b82-de5d-4352-a482-bd662df832d2.png)

En testant sur EcoIndex, j'obtient un poids de page de plus de 3mo ! Ce qui fait entre 1/3 et 1/2 de poids supplémentaire par rapport au reste. 
![BugPoidsEcoIndex](https://user-images.githubusercontent.com/12815769/190845758-a7e4ed42-812a-430e-8073-979bdf682ced.png)

Est-ce que l'on a une idée de pourquoi nous avons tant de différence de poids de page avec EcoIndex ? 
Peut-être que l'on récupère le poids total des ressources après décompression coté navigateur et non pas le poids transférés sur le réseau ? Quelle est la règle en place par rapport à ça ?

**Comment by vvatelot** [comment]
Voilà ce qu'il se passe sur le moteur Ecoindex de l'API:
- On lance un navigateur Chrome sans session
- On logue toutes les données de performance du navigateur (https://github.com/cnumr/ecoindex_scrap_python/blob/main/ecoindex_scraper/scrap.py#L49)
- On charge la page, puis on [attend 3 secondes](https://github.com/cnumr/ecoindex_api#page-wait) : https://github.com/cnumr/ecoindex_scrap_python/blob/main/ecoindex_scraper/scrap.py#L88
- On scrolle en bas de page :  https://github.com/cnumr/ecoindex_scrap_python/blob/main/ecoindex_scraper/scrap.py#L91
- On attend de nouveau 3 secondes : https://github.com/cnumr/ecoindex_scrap_python/blob/main/ecoindex_scraper/scrap.py#L93
- Et seulement ensuite on récupère les métriques de la page

Pour le calcul de données téléchargées, ça se passe ici : https://github.com/cnumr/ecoindex_scrap_python/blob/main/ecoindex_scraper/scrap.py#L124. On récupère toutes les `encodedDataLength` des message de log de type `info` qui comportent `Network.loadingFinished` et on va les additionner. Peut être ai je fait une erreur dans la formule, mais je pense que c'est ça.

Ce scénario permet de simuler un peu plus la vie réelle que de charger la page et la refermer immédiatement.

**Reply by DocRoms** [reply]
Okay, je comprends mieux la différence, en général, les audits de web-performance testent la page qui se charge... ...sans scroll! Ce qui permet de détecter les soucis de cache ou de ressources non lazy dans les résultats affichés.

Néanmoins, et personnellement, j'aime aussi voir le comportement effectué par un utilisateur sur la lecture complète de la page. C'est une bonne chose d'avoir ce scénario. Je penses par contre qu'il faut l'indiquer côté "front". 

![Screenshot_20220917_202815](https://user-images.githubusercontent.com/12815769/190896734-02050c80-3da4-4b9b-8afd-74a3ba87ac9c.jpg)

Ajouter un `après avoir attendu 3 secondes et avoir scrollé en bas de page` après l'explication peu permettre d'éviter les incompréhensions 😊.

D'ailleurs, j'ai l'impression qu'il y a un soucis avec le texte `pour en savoir plus sur notre méthode de calcul`. Il ne sert à rien ou alors il manque un lien. @Bastou @aureliebaton vous savez ce que disent les specs sur ce point ?
